src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -220,10 +245,73 @@ private bool IsConflictMarker(SourceText text, SyntaxTrivia trivia, char ch)
                 text[trivia.SpanStart] == ch;
         }
 
+        private async Task<Document> FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            string equivalenceKey, CancellationToken cancellationToken)
+        {
+            Debug.Assert(
+                equivalenceKey == TakeTopEquivalenceKey ||
+                equivalenceKey == TakeBottomEquivalenceKey ||
+                equivalenceKey == TakeBothEquivalenceKey);
+
+            // Process diagnostics in order so we produce edits in the right order.
+            var orderedDiagnostics = diagnostics.OrderBy(
+                (d1, d2) => d1.Location.SourceSpan.Start - d2.Location.SourceSpan.Start).ToImmutableArray();
```

Comment:
💡 The other option is processing the diagnostics in any order, and then sorting `edits` prior to application. This approach reduces allocations since the sorting will happen on a mutable collection.

Reply:
that's true.  i'm ok with how this is now though giventhat this should stillbe pretty minor in terms of impact, and because i find it much easier to reason and read (and debug) when we're just doing a linear pass.